src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/appointments/AppointmentsService.kt

```diff
@@ -1,39 +1,59 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.appointments
 
 import com.fasterxml.jackson.annotation.JsonFormat
-import com.fasterxml.jackson.annotation.JsonInclude
+import jakarta.validation.constraints.NotEmpty
+import jakarta.validation.constraints.NotNull
+import jakarta.validation.constraints.Size
 import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.awaitBody
-import java.time.LocalDateTime
+import java.time.LocalDate
+import java.time.LocalTime
 
 @Service
 class AppointmentsService(@Qualifier("activitiesApiWebClient") private val webClient: WebClient) {
-  suspend fun createAppointment(createAppointmentRequest: CreateAppointmentRequest): CreateAppointmentResponse =
+  suspend fun createAppointment(appointmentMigrateRequest: AppointmentMigrateRequest): CreateAppointmentResponse =
     webClient.post()
-      .uri("/appointments/TODO")
-      .bodyValue(createAppointmentRequest)
+      .uri("/migrate-appointment")
+      .bodyValue(appointmentMigrateRequest)
       .retrieve()
       .awaitBody()
 }
 
-@JsonInclude(JsonInclude.Include.NON_NULL)
-data class CreateAppointmentRequest(
-  // TODO
-  val bookingId: Long,
-  val offenderNo: String,
-  val prisonId: String? = null,
-  val internalLocation: Long? = null,
-  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
-  val startDateTime: LocalDateTime? = null,
-  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
-  val endDateTime: LocalDateTime? = null,
-  val comment: String? = null,
-  val subtype: String,
-  val status: String,
+data class CreateAppointmentResponse(
+  val id: Long,
 )
 
-data class CreateAppointmentResponse(
-  val appointmentInstanceId: Long,
+data class AppointmentMigrateRequest(
+
+  @field:NotEmpty(message = "Prison code must be supplied")
```

Comment:
How come all of these are nullable types but also have validation for NotEmpty or NotNull?

Reply:
Will do in next iteration